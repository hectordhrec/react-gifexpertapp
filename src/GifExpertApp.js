import React, { useState } from 'react'
import {  } from "module";
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

// export const GifExpertApp = () => { //++Normal
export const GifExpertApp = ({defaultCategories = []}) => { // para las pruebas

    //const [categories, setCategories] = useState(['Rick and Morty']); //++Normal
    const [categories, setCategories] = useState(defaultCategories); // para las pruebas

    // const handleAdd = () => {
    //     // setCategories([...categories, 'South Park']);
    //     /** Otra forma: setCategories recibe un callback donde el primer valor es el estado anterior y luego regresa el nuevo estado*/
    //     setCategories(categories => ['South Park', ...categories]);
    // }

    return (
        <>
            <h1>GiftExpertApp</h1>
            <AddCategory setCategories ={ setCategories } />
            <hr/>

            {/* <button onClick={handleAdd}>Agregar</button> */}

            <ol>{categories.map( category => (
                <GifGrid key={category} category={category} />
            ))}</ol>
        </>
    );
}

