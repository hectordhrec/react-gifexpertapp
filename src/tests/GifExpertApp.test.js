import { shallow } from 'enzyme'
import { GifExpertApp } from '../GifExpertApp'

describe('Probando el componente <GifExpertApp />', () => {
    
    test('Debe renderizarse correctamente', () => {
        
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {
        
        const categories = ['rick and morty', 'Friends'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
    
});
