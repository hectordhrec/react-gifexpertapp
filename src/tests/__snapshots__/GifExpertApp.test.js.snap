// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <GifExpertApp /> Debe renderizarse correctamente 1`] = `
<Fragment>
  <h1>
    GiftExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Probando el componente <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GiftExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="rick and morty"
      key="rick and morty"
    />
    <GifGrid
      category="Friends"
      key="Friends"
    />
  </ol>
</Fragment>
`;
